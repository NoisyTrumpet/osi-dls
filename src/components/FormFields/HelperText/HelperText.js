import React from 'react';
import PropTypes from 'prop-types';
import './HelperText.scss';
import classNames from 'classnames';
import { Typography } from '../../Typography';

const HelperText = ({ addClass, error, children }) => {
  const classes = classNames(
    'helper-text',
    addClass,
    { error },
  );

  return (
    <div className={classes} role={error ? 'alert' : undefined}>
      <Typography variant="helper">
        {children}
      </Typography>
    </div>
  );
};

HelperText.defaultProps = {};

HelperText.propTypes = {
  addClass: PropTypes.string,
  error: PropTypes.bool,
  children: PropTypes.node,
};

export default HelperText;
