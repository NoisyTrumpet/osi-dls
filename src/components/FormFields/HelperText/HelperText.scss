.helper-text {
  margin-top: 6px;
  padding-left: 10px;

  &.error .typography-helper {
    color: map-get($colors, 'util-error');
  }
}

