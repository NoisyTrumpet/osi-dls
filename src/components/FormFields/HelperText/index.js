export { default as HelperText } from './HelperText'; // eslint-disable-line import/prefer-default-export
