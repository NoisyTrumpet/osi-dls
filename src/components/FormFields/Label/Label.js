import React from 'react';
import PropTypes from 'prop-types';
import './Label.scss';
import classNames from 'classnames';
import { Typography } from '../../Typography';

const Label = ({
  fieldType,
  forId,
  display,
  hideLabel,
  isDisabled,
  color,
  required,
  addClass,
  suffixHelperText,
  children,
}) => {
  const classes = classNames(
    'input-label',
    fieldType,
    display,
    hideLabel && 'hide-label',
    addClass,
    { isDisabled },
  );

  const srtVariant = hideLabel && "srt";

  return (
    <label
      className={classes}
      htmlFor={forId}
    >
      <Typography variant={srtVariant || "field-label"} color={color} addClass={addClass}>
        {children}
      </Typography>
      {suffixHelperText && (
        <Typography variant="helper" addClass="optional">
          {suffixHelperText}
        </Typography>
      )}
      {!required && (
        <Typography variant={srtVariant || "helper"} addClass="optional">
          (Optional)
        </Typography>
      )}
    </label>
  );
};

Label.defaultProps = {
  fieldType: 'text',
  display: 'inline',
  isDisabled: false,
  color: '',
};

Label.propTypes = {
  fieldType: PropTypes.oneOf([
    'text',
    'radio',
    'checkbox',
    'textarea',
    'select',
  ]),
  forId: PropTypes.string,
  display: PropTypes.oneOf([
    'inline',
    'inline-block',
    'block',
  ]),
  hideLabel: PropTypes.bool,
  isDisabled: PropTypes.bool,
  color: PropTypes.string,
  suffixHelperText: PropTypes.string, // helper text that is appended to label
  required: PropTypes.bool,
  addClass: PropTypes.string,
  children: PropTypes.node,
};

export default Label;
