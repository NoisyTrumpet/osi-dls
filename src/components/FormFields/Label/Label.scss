.input-label {
  color: map-get($colors, 'gray5');
  padding-left: $space-8;
  order: 2;

  &.block {
    display: block;
  }

  &.inline-block {
    display: inline-block;
  }

  &.text {
    position: absolute;
    top: rem(-11px);
    left: rem(2px);
    transform: translate(0, 50%);

    .optional {
      font-size: rem(12px);
      padding-left: rem(2px);
      padding-right: 0px;
      position: absolute;
      background: #ffffff;
    }
    
    .typography-field-label {
      background: map-get($colors, 'brand-white');
      padding: 0 $space-4;
    }
  }
}
