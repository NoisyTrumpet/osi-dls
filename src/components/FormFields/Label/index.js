export { default as Label } from './Label'; // eslint-disable-line import/prefer-default-export
