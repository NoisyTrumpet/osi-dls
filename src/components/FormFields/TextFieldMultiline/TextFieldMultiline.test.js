import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Formik, Form } from 'formik';
import TextareaAutosize from 'react-textarea-autosize';
import { TextFieldMultiline } from './';
import { Label } from '../Label';
import { HelperText } from '../HelperText';

describe('<TextFieldMultiline />', () => {
  it('Should render', () => {
    const wrapper = mount(
      <Formik initialValues={{ textField: '' }}>
        <Form>
          <TextFieldMultiline
            label="label"
            name="textField"
            domId="textFieldId"
            multiline="false"
            helperText="Optional helper text"
          />
        </Form>
      </Formik>,
    );
    expect(wrapper.find(Label).length).toBe(1);
    expect(wrapper.find(HelperText).length).toBe(1);
    expect(wrapper.find(TextareaAutosize).length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
