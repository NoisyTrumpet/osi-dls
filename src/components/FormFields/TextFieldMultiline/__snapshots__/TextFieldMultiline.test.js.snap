// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextFieldMultiline /> Should render 1`] = `
<Formik
  initialValues={
    Object {
      "textField": "",
    }
  }
>
  <Form>
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <TextFieldMultiline
        domId="textFieldId"
        helperText="Optional helper text"
        iconOnClick={null}
        label="label"
        multiline="false"
        name="textField"
      >
        <div
          className="input-field-container input-field-text"
          data-testid="test-id-input"
        >
          <div
            className="input-field textarea-wrapper"
          >
            <ForwardRef(TextareaAutosize)
              className="input-field-textarea"
              error="false"
              multiline="false"
              name="textField"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            >
              <textarea
                className="input-field-textarea"
                error="false"
                multiline="false"
                name="textField"
                onBlur={[Function]}
                onChange={[Function]}
                value=""
              />
            </ForwardRef(TextareaAutosize)>
          </div>
          <Label
            color="gray4"
            display="block"
            fieldType="text"
            forId="textFieldId"
            isDisabled={false}
          >
            <label
              className="input-label text block"
              htmlFor="textFieldId"
            >
              <Typography
                color="gray4"
                variant="field-label"
              >
                <span
                  className="typography typography-field-label color-gray4"
                >
                  label
                </span>
              </Typography>
              <Typography
                addClass="optional"
                variant="helper"
              >
                <span
                  className="typography typography-helper optional"
                >
                  (Optional)
                </span>
              </Typography>
            </label>
          </Label>
          <HelperText>
            <div
              className="helper-text"
            >
              <Typography
                variant="helper"
              >
                <span
                  className="typography typography-helper"
                >
                  Optional helper text
                </span>
              </Typography>
            </div>
          </HelperText>
        </div>
      </TextFieldMultiline>
    </form>
  </Form>
</Formik>
`;
